crates/router/src/core/payment_methods/cards.rs

```diff
@@ -3277,11 +3502,16 @@ pub async fn set_default_payment_method(
         .await
         .change_context(errors::ApiErrorResponse::InternalServerError)
         .attach_printable("Failed to update the default payment method id for the customer")?;
+
+    let pm = payment_method
```

Comment:
payment method can be None in this case right?

Reply:
Since payment method is a required field here, we'll throw an error here in case it is None.

Reply:
you should move this before updating the customer details